cocos/core/gfx/webgl2/webgl2-commands.ts

```diff
@@ -1119,13 +1119,30 @@ export function WebGL2CmdFuncCreateTexture (device: WebGL2Device, gpuTexture: IW
 }
 
 export function WebGL2CmdFuncDestroyTexture (device: WebGL2Device, gpuTexture: IWebGL2GPUTexture) {
+    const { gl } = device;
     if (gpuTexture.glTexture) {
-        device.gl.deleteTexture(gpuTexture.glTexture);
+        const glTexUnits = device.stateCache.glTexUnits;
+        let texUnit = 0;
```

Comment:
This will again introduce cache incoherencies when none of the texture units currently store the texture to be deleted. Should init this to `device.stateCache.texUnit`